commit 924531326e2dd4ceabe7240f2b55a88e7d894ec2
Author: Felix Fietkau <nbd@nbd.name>
Date:   Thu Mar 30 14:08:40 2023 +0200

    net: ethernet: mtk_eth_soc: add missing ppe cache flush when deleting a flow
    
    The cache needs to be flushed to ensure that the hardware stops offloading
    the flow immediately.
    
    Fixes: 33fc42de3327 ("net: ethernet: mtk_eth_soc: support creating mac address based offload entries")
    Reviewed-by: Simon Horman <simon.horman@corigine.com>
    Signed-off-by: Felix Fietkau <nbd@nbd.name>
    Reviewed-by: Leon Romanovsky <leonro@nvidia.com>
    Link: https://lore.kernel.org/r/20230330120840.52079-3-nbd@nbd.name
    Signed-off-by: Jakub Kicinski <kuba@kernel.org>

--- a/drivers/net/ethernet/mediatek/mtk_ppe.c
+++ b/drivers/net/ethernet/mediatek/mtk_ppe.c
@@ -414,6 +414,7 @@ __mtk_foe_entry_clear(struct mtk_ppe *pp
 		ppe->foe_table[entry->hash].ib1 |= FIELD_PREP(MTK_FOE_IB1_STATE,
 							      MTK_FOE_STATE_INVALID);
 		dma_wmb();
+		mtk_ppe_cache_clear(ppe);
 	}
 	entry->hash = 0xffff;
 
